src/net_processing.cpp

```diff
@@ -2252,7 +2250,7 @@ void ProcessMessage(
         {
             connman->SetServices(pfrom.addr, nServices);
         }
-        if (!pfrom.fInbound && !pfrom.fFeeler && !pfrom.m_manual_connection && !HasAllDesirableServiceFlags(nServices))
+        if ((pfrom.m_conn_type == ConnectionType::OUTBOUND || pfrom.m_conn_type == ConnectionType::BLOCK_RELAY || pfrom.m_conn_type == ConnectionType::ADDR_FETCH) && !HasAllDesirableServiceFlags(nServices))
```

Comment:
Consider using a `switch` without `default:` to assign to a boolean variable the expression `type == FOO || type == BAR || type == BAZ` to enforce the code to be reassessed when new enum element is added.

Reply:
extracted into `ExpectServicesFromConn` & handled as switch statement